python/ray/utils.py

```diff
@@ -422,6 +444,16 @@ def estimate_available_memory():
         overestimate if psutil is not installed.
     """
 
+    # check cgroup memory first
+    try:
+        with open("/sys/fs/cgroup/memory/memory.usage_in_bytes", "rb") as f:
```

Comment:
@edoakes will this also help in the kubernetes case?

Reply:
Yes, this will certainly help in k8s. The other option in k8s is to pass the allocated memory into the initial ray command using the k8s-provided environment variable, but it would be nicer to have it "just work."

Reading the cgroup pseudo-files seems fine, there are only two minor concerns to keep in mind: (1) we might want to support other isolation mechanisms in the future (e.g., firecracker VMs, kata containers) and (2) this API will change if/when docker migrates to cgroups v2.